frontend/public/components/start-guide.tsx

```diff
@@ -10,39 +10,27 @@ import { ProjectModel, RoleModel, StorageClassModel } from '../models';
 
 const WHITELIST = [RoleModel.kind, StorageClassModel.kind];
 
-const seenGuide: string = 'seenGuide';
-
-export class StartGuide extends React.Component<StartGuideProps, StartGuideState> {
-  constructor(props) {
-    super(props);
-
-    // TODO: The dismissable logic is not currently being used, but we'll
-    // probably want a general start guide for OpenShift separate from the no
-    // projects message. Leaving this for now.
-    this.dismiss = this.dismiss.bind(this);
-    let {visible} = props;
-    try {
-      visible = visible || !localStorage.getItem(seenGuide);
-    } catch (ignored) {
-      // ignored
-    }
-    this.state = {
-      visible,
-    };
+export const StartGuide: React.FC<StartGuideProps> = (props) => {
+  let {localStorageKey, startGuide, visible} = props;
+  const seenGuide: string = localStorageKey ? localStorageKey : 'seenGuide';
+  try {
+    visible = visible || !localStorage.getItem(seenGuide);
+  } catch (ignored) {
+    // ignored
   }
+  const [isVisible, setIsVisible] = React.useState(visible);
 
-  dismiss() {
-    this.setState({
-      visible: false,
-    });
+  const dismiss = () => {
+    setIsVisible(false);
     localStorage.setItem(seenGuide, 'true');
-  }
+  };
 
-  render() {
-    const { visible } = this.state;
-    return visible ? <OpenShiftGettingStarted /> : null;
-  }
-}
+  const guide = startGuide ? React.cloneElement(startGuide, { dismiss: () => dismiss() }) : <OpenShiftGettingStarted />;
```

Comment:
Don't have a default. Just pass in `OpenShiftGettingStarted` when that's the guide we want.